pkg/asset/machines/openstack/machinesets.go

```diff
@@ -46,9 +51,11 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 			return nil, err
 		}
 
-		// TODO(flaper87): Implement AZ support sometime soon
-		//name := fmt.Sprintf("%s-%s-%s", clustername, pool.Name, az)
-		name := fmt.Sprintf("%s-%s", clusterID, pool.Name)
+		if az != "" {
+			azString = fmt.Sprintf("-%s", az)
+		}
+
+		name := fmt.Sprintf("%s-%s%s", clusterID, pool.Name, azString)
```

Comment:
what about dropping `azString` altogether

```Go
name := clusterID + "-" + pool.Name
if az != "" {
	name += "-" + az
}
```

Reply:
The naming issue was addressed separately in #4000. Can you update your patch leaving the naming issue aside?